### PR TITLE
Fixing FLASHParticleFile IO

### PR DESCRIPTION
**Pull request by matthewturk**
This implements `_yield_coordinates` for the `FLASHParticleDataset` type, which is a blocker for running the tests.

@jzuhone this look reasonable to you?

### CODE REVIEW

**Comment by jzuhone** [comment]
@matthewturk I actually worked on this a bit last week:

https://github.com/jzuhone/yt/commit/143807491734a9406f62193b6dee057a82d43935

Your code is cleaner, but I still had some issues which I wasn’t able to figure out.

**Comment by matthewturk** [comment]
I don't mind pulling your changeset in instead, if you'd rather!

**Comment by jzuhone** [comment]
You can--you should check my work regarding `pcount`. Also, I was getting failures in the `assert(len(data_files) == 1)` lines.

**Comment by matthewturk** [comment]
@jzuhone `len(data_files)==1` will only be true if we do what gets done in the `_read_particle_fields` call, which is to make `data_files` be a set of file names.  That applies to sub-chunking, which we (explicitly) don't need to do in `_yield_coordinates`.

**Comment by jzuhone** [comment]
@matthewturk ok--so how should we handle things?

File: yt/frontends/flash/io.py [hunk 1]
```diff
@@ -183,6 +183,12 @@ def _read_particle_coords(self, chunks, ptf):
                     total += count
                     yield ptype, (x, y, z)
 
+    def _yield_coordinates(self, data_file, needed_ptype=None):
+        px, py, pz = self._position_fields
+        p_fields = self._handle["/tracer particles"]
+        pxyz = np.asarray(p_fields[data_file.start:data_file.end], dtype="=f8")
+        yield ("io", pxyz)
```
Comment (jzuhone) [inline yt/frontends/flash/io.py:190]: I don't think this does what you want--I think it grabs all of the fields instead of only the position ones.

Comment (matthewturk) [reply yt/frontends/flash/io.py:190]: HA!  You're right.  Before my correction for flake8, it was `p_fields[data_file.start:data_file.end,(px,py,pz)]`.

**Comment by matthewturk** [comment]
@jzuhone Do you have any changes to other stuff in that branch of development?  If so, I'll go ahead and merge them both together, but if the `_yield_coordinates` function I wrote is OK I'll keep it.

**Comment by jzuhone** [comment]
@matthewturk with the fix we just discussed, your `_yield_coordinates` is probably preferable. It would probably be nice to have a function that creates smoothing lengths similar to what is done for tipsy, but I think that's probably out of the scope of this PR because if we're going to use it for multiple frontends we should probably do something more broad.

**Comment by matthewturk** [comment]
@jzuhone ok -- I looked it over and what I'm going to do is merge from your branch and address the changes (including whitespace and `_count_particles`) then push to this PR.

**Comment by matthewturk** [comment]
Need to fix one more thing, but am away for a few.

File: yt/frontends/flash/io.py [hunk 2]
```diff
@@ -170,18 +172,22 @@ def _read_particle_coords(self, chunks, ptf):
...
-                total = 0
-                while total < pcount:
-                    count = min(self._chunksize, pcount - total)
-                    x = np.asarray(p_fields[total:total+count, px], dtype="=f8")
-                    y = np.asarray(p_fields[total:total+count, py], dtype="=f8")
-                    z = np.asarray(p_fields[total:total+count, pz], dtype="=f8")
-                    total += count
-                    yield ptype, (x, y, z)
+            pxyz = np.asarray(p_fields[data_file.start:data_file.end, (px,py,pz)], dtype="=f8")
+            yield "io", pxyz
```
Comment (matthewturk) [inline yt/frontends/flash/io.py:184]:
```suggestion
            yield "io", pxyz.T
```

Comment (jzuhone) [reply yt/frontends/flash/io.py:184]: I suspect that should get you the right shape?

Comment (matthewturk) [reply yt/frontends/flash/io.py:184]: yup, am fixing `_read_particle_fields` now

**Comment by matthewturk** [comment]
@jzuhone I am merging this in to my `yt4-merge` work, but I need your advice -- now, a projection of `("deposit", "all_density")` throws an error.  What should I replace that with?

**Comment by jzuhone** [comment]
@matthewturk I don't think we can replace it with anything, actually? these particles don't have smoothing lengths, so there's nothing to deposit or project. The assumption with these files is that there is no grid to speak of.

**Comment by matthewturk** [comment]
Alright, that sounds good to me.  I'll yank those tests in the other PR.

File: yt/frontends/flash/io.py [hunk 3]
```diff
@@ -170,18 +172,15 @@ def _read_particle_coords(self, chunks, ptf):
...
-                    total += count
-                    yield ptype, (x, y, z)
+            pxyz = np.asarray(p_fields[data_file.start:data_file.end, (px,py,pz)], dtype="=f8")
+            yield "io", pxyz.T
+
+    def _yield_coordinates(self, data_file, needed_ptype=None):
+        px, py, pz = self._position_fields
+        p_fields = self._handle["/tracer particles"]
+        pxyz = np.asarray(p_fields[data_file.start:data_file.end, (px,py,pz)], dtype="=f8")
+        yield ("io", pxyz)
```
Comment (munkm) [inline yt/frontends/flash/io.py:183]: is there a reason this is `yield ("io", pxyz)` and not `yield "io", pxyz`?

**Comment by munkm** [comment]
So these commits somehow got incorporated into the 4.0 merge PR and already exist on master now.

**Pull request closed without merging**